include/gridtools/sid/simple_ptr_holder.hpp

```diff
@@ -29,15 +30,10 @@ namespace gridtools {
         GT_TARGET_NAMESPACE {
             template <class T>
             struct simple_ptr_holder {
-                T m_val;
+                ldg_ptr_t<T> m_val;
 
-#if defined(__cpp_deduction_guides) and __cpp_deduction_guides >= 201907
```

Comment:
I have this PR open https://github.com/GridTools/gridtools/pull/1775 related to this change. If I remember correctly, on broken compilers (12.4 and earlier) you get more stuff to work if you use C++20 and the constructor is not define, i.e. ctad for aggregates works.

Reply:
OK, I will move the wrapping with `as_ldg_ptr` to `operator()`. Then everything should work as before.

Reply:
Ah, I missed that ldg_ptr is not an aggregate. So ok, to keep I guess. But also could you make ldg_ptr an aggregate with the same macro, no?